<?xml version="1.0"?>
<launch>
    <arg name="hz" default="10"/>
    <arg name="epsilon" default="1.0"/>
    <arg name="max_speed" default="1.0"/>
    <arg name="max_yawrate" default="1.0"/>
    <arg name="kernel_size" default="20"/>
    <arg name="max_range" default="10.0"/>
    <arg name="max_dis" default="10.0"/>
    <arg name="model_path" default="/home/amsl/ros_catkin_ws/src/mpdrl_ros/model/libtorch_model/rnn_ppo_model.pt"/>

    <node name="localmap2scan" pkg="mpdrl_ros" type="localmap_scan">
        <remap from="/local_map" to="/local_costmap"/>
        <remap from="/local_map/scan" to="/local_map/scan"/>
    </node>

    <node name="mpdrl" pkg="mpdrl_ros" type="mpdrl" output="screen">
        <param name="HZ" value="$(arg hz)" type="int"/>
        <param name="EPSILON" value="$(arg epsilon)" type="double"/>
        <param name="MAX_SPEED" value="$(arg max_speed)" type="double"/>
        <param name="MAX_YAWRATE" value="$(arg max_yawrate)" type="double"/>
        <param name="KERNEL_SIZE" value="$(arg kernel_size)" type="int"/>
        <param name="MAX_RANGE" value="$(arg max_range)" type="double"/>
        <param name="MAX_DIS" value="$(arg max_dis)" type="double"/>
        <param name="MODEL_PATH" value="$(arg model_path)" type="str"/>
        <remap from="/scan" to="/local_map/scan"/>
        <remap from="/cmd_vel" to="/cmd_vel"/>
    </node>
</launch>
